sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SqlParser.scala

```diff
@@ -198,6 +203,44 @@ class SqlParser extends AbstractSparkSQLParser with DataTypeParser {
     | FULL  ~ OUTER.? ^^^ FullOuter
     )
 
+  protected lazy val groupType: Parser[(LogicalPlan, Seq[NamedExpression]) => LogicalPlan] =
+    GROUP ~ BY ~> (
+      rep1sep(expression, ",") <~ WITH ~ ROLLUP ^^ { case r =>
+        (child: LogicalPlan, aggregations: Seq[NamedExpression]) =>
+          Rollup(r, child, aggregations)
+      }
+      | rep1sep(expression, ",") <~ WITH ~ CUBE ^^ { case c =>
+          (child: LogicalPlan, aggregations: Seq[NamedExpression]) =>
+            Cube(c, child, aggregations)
+      }
+      | rep1sep(expression, ",") ~ (GROUPING ~ SETS ~ "(" ~> rep1sep(groupingsets, ",") <~ ")")
+          ^^ { case e ~ g =>
+          (child: LogicalPlan, aggregations: Seq[NamedExpression]) =>
+            calGroupingSets(child, aggregations, g, Some(e))
+      }
+      | rep1sep(expression, ",") ^^ { case g =>
+          (child: LogicalPlan, aggregations: Seq[NamedExpression]) =>
+            Aggregate(g, aggregations, child)
+      }
+    )
+
+  protected lazy val groupingsets: Parser[Seq[Expression]] =
+    ( "(" ~> repsep(expression, ",") <~ ")"
+    | rep1sep(expression, ",")
+    )
+
+  private def calGroupingSets(child: LogicalPlan, aggregations: Seq[NamedExpression],
```

Comment:
It's because I want to compat syntax `GROUP BY GROUPING SETS(expression list)` in https://github.com/apache/spark/pull/5045 before, and I will change to `groupByExprs: Seq[Expression]` in this patch, thanks for review the code.